developer-docs-site/docs/guides/nfts/mint-nft-v2-cli.md

```diff
@@ -0,0 +1,918 @@
+:::
+
+Let's start by adding the `SignerCapability` to our contract, which is the structure that produces the capability to sign a transaction programmatically.
+
+We'll store it in our `MintConfiguration`:
+
+```rust
+struct MintConfiguration has key {
+    signer_capability: SignerCapability,
+    collection_name: String,
+    token_name: String,
+    token_uri: String,
+}
+```
+
+We create it, providing it a seed in the form of the collection name.
+
+```rust
+let seed = *string::bytes(&collection_name);
+let (resource_signer, resource_signer_cap) = account::create_resource_account(owner, seed);
+```
+
+:::info
+The seed can be anything we want, but since resource accounts are unique hashes of the combination of the creating account + seed, it's good to make the seed something that will also be unique. In our case, the owner and collection name combination will always be unique because that is a constraint enforced by the `collection.move` contract, so the seed being the collection name logically follows.
+:::
+
+To clarify, the `resource_signer` is the actual structure on chain that signs things, it is of type `signer`; whereas the `SignerCapability` is a unique
+on-chain resource that generates a signer for an account, given whomever requesting it has permission to access the `SignerCapability` resource.
+
+Now we can provide the `resource_signer` as the creator of the collection, and move the `resource_signer_cap` to the `MintConfiguration`, so we can programmatically retrieve the creator's ability to sign later.
+
+Note also that we store the `MintConfiguration` resource onto the resource account now, so when you call `create_collection` you'll have to look up the resource account's address to call the mint function later.
+
+```rust
+aptos_token::create_collection(
+    &resource_signer,
+    // ...
+);
+
+move_to(&resource_signer, MintConfiguration {
+    signer_capability: resource_signer_cap,
+    collection_name,
+    token_name,
+    token_uri,
+});
+```
+
+Let's alter the mint function so that it uses the resource account instead of the owner account.
+
+The first thing to notice is that the arguments to the function have changed. We no longer need the owner or the creator to sign the transaction. To do this before, we would've had to implement a function that takes two signers, which would've been complex. Not requiring the signer, however, meant the receiver had no say in whether or not they even wanted to receive the NFT.
+
+Now, we can require the receiver to sign so that a user can mint whenever they like, and the owner doesn't have to approve of it beforehand.
+
+```rust
+public entry fun mint(receiver: &signer, resource_addr: address) acquires MintConfiguration {
+    //...
+}
+```
+
+Note that we require the user to pass in the resource address- we've provided a view function as one way for you to calculate it. We show you how to use this function later in the [Running the contract section](#running-the-contract-1).
```

Comment:
I wonder if we can just deploy the module at the resource account instead. This would remove the requirement that the user needs to pass in the resource account address

Reply:
Yeah, but I was trying to move away from the flow of publishing modules to resource accounts because of how difficult it is to upgrade. When I was doing the tutorial myself it was especially frustrating when I screwed anything up and had to re-deploy to a new account because of a typo or because I messed with something, haha

I also considered just storing it in a resource, but then I realized whenever I actually used a minting contract in an e2e app I always ended up generating the resource address and storing it in state at some point, so it didn't matter, especially with an easy `#[view]` function for it

Reply:
It's not a great UX IMO that the user needs to pass the resource account address every time. Upgrading is not great with resource accounts currently for sure but (1) that's a problem for another tutorial :) and (2) we can improve the flow (it's part of our Q2 Move roadmap actually)

Reply:
Ya I agree, I may have been overthinking it in this case, but my thought process was the above + there's a comment right above the function call to generate a resource address that says:

>  /// This is a helper function to compute resource addresses. Computation of the address
>  /// involves the use of a cryptographic hash operation and should be use thoughtfully.

So I tried to avoid computing it for every single mint transaction.

I was going to store it on contract account itself, which in this case makes sense, but I didn't want to encourage a bad habit of storing resources there.

In this case, it probably makes the most sense to store it on the owner's account, which in this case is the contract account anyway, yea?

Reply:
I mean isn't this more akin to minting from an object?

We're passing in our mint resource?

Also have we considered using an object instead of a resource account? It seems to be a more sensible approach ... considering.